app/Http/Controllers/Legacy/Web/CampaignsController.php

```diff
@@ -78,4 +97,66 @@ public function destroy(Campaign $campaign)
 
         // @TODO: redirect to campaign deleted page
     }
+
+    /**
+     * Create a new campaign.
+     */
+    public function create()
+    {
+        return view('pages.campaigns_create');
+    }
+
+    /**
+     * Show a specific campaign page.
+     *
+     * @param  \Rogue\Models\Campaign  $campaign
+     */
+    public function show(Campaign $campaign)
+    {
+        // Format start and end dates how we want them to be viewed.
+        $campaign = $this->formatStartEndDates($campaign, 'There is no end date for this campaign.');
+
+        return view('pages.campaigns_show')->with('campaign', $campaign);
+    }
+
+    /**
+     * Edit a specific campaign page.
+     *
+     * @param  \Rogue\Models\Campaign  $campaign
+     */
+    public function edit(Campaign $campaign)
+    {
+        // Format start and end dates how we want them to be viewed.
+        $campaign = $this->formatStartEndDates($campaign, null);
+
+        return view('pages.campaigns_edit')->with('campaign', $campaign);
+    }
+
+    /**
+     * Helper function to format start and end dates for frontend.
+     *
+     * @param $campaign
+     * @param $end_date_placeholder
+     */
+    public function formatStartEndDates($campaign, $end_date_placeholder)
+    {
+        $campaign->start_date = date("m/d/Y", strtotime($campaign->start_date));
```

Comment:
I'm wondering if the piece of this function where we get the display date should live on the model? We have something similar with [getting a user's display name in Northstar](https://github.com/DoSomething/northstar/blob/master/app/Models/User.php#L335-L349)

Reply:
I love that idea @katiecrane - updated!